CMakeLists.txt

```diff
@@ -231,12 +231,17 @@ IF(NOT STATIC_LINK)
 ENDIF()
 
 # Check for all the required and optional dependencies
-FIND_PACKAGE(PythonLibs 2.3 REQUIRED)
-IF(PYTHONLIBS_FOUND)
-	MESSAGE(STATUS "Looking for Python libraries and headers: found")
+IF(NOT AmigaOS4)
+	FIND_PACKAGE(PythonLibs 2.3 REQUIRED)
+	IF(PYTHONLIBS_FOUND)
+		MESSAGE(STATUS "Looking for Python libraries and headers: found")
+	ELSE()
+		MESSAGE(SEND_ERROR "Looking for Python libraries and headers: not found!")
+		MESSAGE(FATAL_ERROR "Please get them (www.python.org)")
+	ENDIF()
 ELSE()
-	MESSAGE(SEND_ERROR "Looking for Python libraries and headers: not found!")
-	MESSAGE(FATAL_ERROR "Please get them (www.python.org)")
+	# Due a bug cross compiler environment we have to hack this
+        INCLUDE_DIRECTORIES(/usr/ppc-amigaos/SDK/local/newlib/include/python2.5/)
```

Comment:
It would be cleaner if you set PYTHONLIBS_FOUND and PYTHON_INCLUDE_DIRS. Surely you also need PYTHON_LIBRARIES so it links properly?

Also, can you invert this check? When you have both if and else, it's an antipattern to start with if(not).

Reply:
Sorry. How do you suggest to do this? Unlucky FIND_PACKAGE(PythonLibs 2.3 REQUIRED) doesn't work even if i specify both PYTHON_INCLUDE_DIRS and PYTHON_LIBRARIES. So where i should set PYTHON_LIBRARIES ?

Reply:
I meant those 2-3 as a replacement for the one INCLUDE_DIRECTORIES, not calling FIND_PACKAGE. Btw, is the problem that it doesn't find it there?

Reply:
ahah.. funny.. the problem is that error.. the correct defines are PYTHON_INCLUDE_**DIR** and PYTHON_**LIBRARY** so i can revert the change